util/container/image_test.go

```diff
@@ -0,0 +1,185 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package container
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/docker/distribution/reference"
+	. "github.com/onsi/gomega"
+)
+
+func TestParseImageName(t *testing.T) {
+	testCases := []struct {
+		name      string
+		input     string
+		repo      string
+		imageName string
+		tag       string
+		digest    string
+		wantError bool
+	}{
+		{
+			name:      "input with path and tag",
+			input:     "k8s.gcr.io/dev/coredns:1.6.2",
+			repo:      "k8s.gcr.io/dev",
+			imageName: "coredns",
+			tag:       "1.6.2",
+			wantError: false,
+		},
+		{
+			name:      "input with name only",
+			input:     "example.com/root",
+			repo:      "example.com",
+			imageName: "root",
+			wantError: false,
+		},
+		{
+			name:      "input with name and tag without path",
+			input:     "example.com/root:tag",
+			repo:      "example.com",
+			imageName: "root",
+			tag:       "tag",
+			wantError: false,
+		},
+		{
+			name:      "input with name and digest without tag",
+			input:     "example.com/root@sha256:e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855",
+			repo:      "example.com",
+			imageName: "root",
+			digest:    "sha256:e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855",
+			wantError: false,
+		},
+		{
+			name:      "input with path and name",
+			input:     "example.com/user/repo",
+			repo:      "example.com/user",
+			imageName: "repo",
+			wantError: false,
+		},
+		{
+			name:      "input with path, anme and tag",
+			input:     "example.com/user/repo:tag",
+			repo:      "example.com/user",
+			imageName: "repo",
+			tag:       "tag",
+			wantError: false,
+		},
+		{
+			name:      "input with path, name, tag and digest",
+			input:     "example.com/user/repo@sha256:e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855",
+			repo:      "example.com/user",
+			imageName: "repo",
+			digest:    "sha256:e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855",
+			wantError: false,
```

Comment:
Can we have an image with tag and digest specified? This test doesn't have a tag specified.